subworkflows/local/generate_downstream_samplesheet/main.nf

```diff
@@ -0,0 +1,52 @@
+//
+// Subworkflow with functionality specific to the nf-core/createtaxdb pipeline
+//
+
+workflow SAMPLESHEET_DIFFERENTIALABUNDANCE {
+    take:
+    ch_taxpasta
+
+    main:
+    format_sep = '\t'
+
+    ch_taxpasta.map { it ->
+        def tool_name = it[0]['tool']
+        def id = it[0]['id']
+        def file_path = it[1]
+        def samplesheet_name = file(file_path).getName()
+
+        ch_list_for_samplesheet = Channel
+            .fromPath(file_path)
+            .splitCsv(sep: format_sep)
+            .map { row -> row.drop(1) }
+            .flatten()
+
+        ch_colnames = Channel.of('sample')
+
+        channelToSamplesheet(ch_colnames, ch_list_for_samplesheet, "downstream_samplesheets/differentialabundance", "samplesheet_${samplesheet_name}" )
```

Comment:
Please see the latest function as in detaxizer! It's tweaked again (sorry about that!)